samples/SocialSample/Startup.cs

```diff
@@ -136,7 +136,7 @@ k web
                         JObject user = JObject.Parse(text);
 
                         var identity = new ClaimsIdentity(
-                            context.Options.AuthenticationType,
+                            context.Options.AuthenticationScheme,
```

Comment:
Barry, It is not disingenuous. The runtime should not be adding anything to the ClaimsPrincipal after the AH has completed it's work. How a ClaimsIdentity is constructed has a direct relationship to ClaimsPrincipal.IsInRole being true or false.

Perhaps, I am not understanding the relationship between cookie versions and claims transformation. What I am looking for is a guarantee that a V1 host can understand a V2 cookie (assuming the same key and algorithm is used for protection).